packages/kbn-user-profile-components/src/user_profiles_selectable.tsx

```diff
@@ -208,18 +227,30 @@ export const UserProfilesSelectable = <Option extends UserProfileWithAvatar | nu
       values.map((option) => {
         if (selectedOptions) {
           const match = selectedOptions.find((profile) => isMatchingOption(option, profile));
-          return { ...option, checked: match === undefined ? undefined : 'on' };
+          const checked = match === undefined ? undefined : 'on';
+          const disabled = checked ? false : limitReached;
+          return {
+            ...option,
+            checked,
+            disabled,
+            prepend: option.data ? (
+              <UserAvatar
+                user={option.data.user}
+                avatar={option.data.data?.avatar}
+                size="s"
+                isDisabled={disabled}
+              />
+            ) : undefined,
```

Comment:
It's not entirely clear to me why you've moved the `UserAvatar` up from the `toSelectableOption` function? But I assume it's as intended 👍

Reply:
I need to render it as disabled under certain conditions and don't have access to that information in `toSelectableOption`